sphinx_needs/directives/needuml.py

```diff
@@ -208,9 +245,22 @@ def process_needuml(app, doctree, fromdocname):
         mem_template = Environment(loader=BaseLoader).from_string(uml_content)
 
         # Get all needed Jinja Helper Functions
-        jinja_utils = JinjaFunctions(app, fromdocname)
+        jinja_utils = JinjaFunctions(app, fromdocname, parent_need_id)
         # Make the helpers available during rendering
-        data = {"needs": all_needs, "uml": jinja_utils.uml, "flow": jinja_utils.flow, "filter": jinja_utils.filter}
+        if current_needuml["is_arch"]:
+            # Add jinja function import() only for needarch
+            data = {
```

Comment:
Define `data` only once and add "import" only if `is_arch`.
Currently, we define the data twice with nearly the same content, hard to maintain.